tests/test.VmAgentRealVmadm.js

```diff
@@ -841,6 +842,136 @@ test('Real vmadm, fake VMAPI: validate DNI', function _test(t) {
 });
 
 /*
+ * This test is designed to validate the correct handling of the
+ * do_not_inventory flag when creating a new DNI VM before or after the VmAgent
+ * is started. This is done by:
+ *
+ *  - creating a VM with do_not_inventory
+ *  - starting VmAgent and allowing the fake VMAPI to fill with the VMs from
+ *    this system (but not the do_not_inventory VM)
+ *  - creating a 2nd VM with do_not_inventory
+ *  - deleting both the VMs with do_not_inventory
+ *  - confirm that no events have been sent to VMAPI for these DNI VMs
+ *  - cleanup
+ *
+ */
+test('Real vmadm, fake VMAPI: new DNI VM', function _test(t) {
+    var config = newConfig();
+    var payload = {
+        autoboot: true,
+        brand: 'joyent-minimal',
+        do_not_inventory: true,
+        image_uuid: smartosImageUUID
+    };
+    var payload1 = JSON.parse(JSON.stringify(payload));
+    var payload2 = JSON.parse(JSON.stringify(payload));
+    var vmAgent;
+
+    payload1.uuid = node_uuid.v4();
+    payload1.alias = 'vm-agent_testvm1';
+    payload1.log = config.log;
+    payload2.uuid = node_uuid.v4();
+    payload2.alias = 'vm-agent_testvm2';
+    payload2.log = config.log;
+
+    vasync.pipeline({arg: {}, funcs: [
+        function _createDoNotInventoryVm1(arg, cb) {
+            // creating a VM with DNI should not result in it being in the
+            // initial update.
+            vmadm.create(payload1, function _vmadmCreateCb(err, info) {
+                t.ifError(err, 'create DNI VM1');
+                if (!err && info) {
+                    t.ok(info.uuid, 'new DNI VM1 has uuid: ' + info.uuid);
+                }
+                cb(err);
+            });
+        }, function _startVmAgent(arg, cb) {
+            t.ok(config.server_uuid, 'new CN ' + config.server_uuid);
+            vmAgent = new VmAgent(config);
+            vmAgent.start();
+            cb();
+        }, function _waitInitialUpdateVms(arg, cb) {
+            // Wait for VmAgent init and it'll send the initial PUT /vms, these
+            // are real VMs on the node because we're not faking vmadm.
+            coordinator.once('vmapi.updateServerVms',
+                function _onUpdateVms(vmobjs /* , server_uuid */) {
+                    var keys = Object.keys(vmobjs);
+
+                    t.ok(true, 'saw PUT /vms: (' + keys.length + ')');
+                    keys.forEach(function _putVmToVmapi(vm) {
+                        // ignore updates from VMs that existed when we started
```

Comment:
You're right, this is actually not needed for this test, since we're not looking at what's in VMAPI here.